src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+/**
+ * Represents the controller to parse and write to a save file for tasks
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * String representation of the file path to the save file
+     * 
+     * @param filePath
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Writes the tasks in task list to the save file
+     * 
+     * @param store
+     *            Task list
+     */
+    public void storeToFile(ArrayList<Task> store) {
+        String text = "";
+        for (int i = 0; i < store.size(); i++) {
+            Task currentTask = store.get(i);
+            text = text + currentTask.toString() + System.lineSeparator();
+        }
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            fw.write(text);
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Parses the save file
+     * 
+     * @return ArrayList of tasks
+     */
+    public ArrayList<Task> fileToStore() {
+        ArrayList<Task> store = new ArrayList<>();
+        File f = new File(filePath);
+        try {
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String currentLine = s.nextLine();
+                String[] arrayElements = currentLine.split("\\|");
+                String taskType = arrayElements[0].trim();
+                Task newTask;
+                switch (taskType) {
+                case "T":
+                    newTask = new ToDo(arrayElements[2].trim(), arrayElements[1].trim().equals("1"));
```

Comment:
Avoid complicated expressions. This can be broken into multiple lines with different methods (i.e. SLAP more). For example, `arrayElements[1].trim().equals("1")` can be refactored to another method.